jetty-ee10/jetty-ee10-webapp/src/main/resources/org/eclipse/jetty/ee10/webapp/catalog.xml

```diff
@@ -0,0 +1,110 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<catalog xmlns="urn:oasis:names:tc:entity:xmlns:xml:catalog" xml:base="jetty-loader:jakarta/servlet/">
```

Comment:
It feels wrong to need to specify `jetty-loader:` in this file.  
If the servlet spec ultimately did include a catalog, surely it would just be a map of publicId/systemId to a uri path within the same jar?

So should it not be the resolver that knows how to turn "resources/web-app_2_2.dtd" into a `Some.class.getResource("resources/web-app_2_2.dtd");`?

Reply:
Can't we do that now?   Use a catalog to convert a publicId/SystemId to a simple URI path, and then use our own resolver that knows how to turn that URI path into a reference into the specific class resource?

Reply:
The `Catalog` is just an interface, we could just provide the `Catalog` implementation as an internal representation too.
It's a trivial class, I bet I can knock that out easily enough.
The base uri can then be whatever we want it to be, and not hardcoded into the XML (like it's spec'd to be)

It's the `CatalogResolver` that has lions share of effort, and that's just using the `Catalog` as provided.

Reply:
> So should it not be the resolver that knows how to turn "resources/web-app_2_2.dtd" into a `Some.class.getResource("resources/web-app_2_2.dtd");`?

Not quite, as per JPMS rules, we cannot use something like `Servlet.class.getResource("resources/web-app_2_2.dtd")` easily enough, that would require the Servlet jar to expose the `jakarta/servlet/resources` tree to us.

The `Loader.getResource("jakarta/servlet/resources/web-app_2_2.dtd")` usage allowed us to get around this (for now, but who knows what kind of new rules will creep up in the future).

Reply:
But the module-info for the servlet API does have `opens jakarta.servlet.resources;`, so we should be able to access it?

Eitherway, the concept as I see it should be that catalog.xml transforms the ID into a local URI and then we can implement a resolver that knows how to get that URI out of a local jar file.  I think `Servlet.class.getResource("jakarta/servlet/resources/web-app_2_2.dtd")` should work, but if not then we use whatever JPMS approved method necessary.  But the point being, is that we don't need to have a `jetty-loader:` scheme in the XML (and probably not at all).

Essentially, we can use the catalog mechanism simple to replace the maps of ID that we were previously maintaining ourselves.